### ISSUE TITLE
DateTime::setTimestamp() doesn't keep microseconds like in PHP

### ISSUE THREAD

**Issue by jesseschalken** [facebook/hhvm#7394]
### HHVM Version

HipHop VM 3.15.0 (rel)
Compiler: tags/HHVM-3.15.0-0-g92a682ebaa3c85b84857852d8621f528607fe27d
Repo schema: 225d4323575bbc8a498dc809a1c41354f6bca830
### Standalone code, or other way to reproduce the problem

``` php
<?php

date_default_timezone_set('UTC');

$t1 = 1493352139;
$t2 = (int)($t1 / 2);

$dt = \DateTime::createFromFormat('U.u', $t1 . '.836264');

print $dt->format('Y-m-d H:i:s.u') . "\n";

$dt->setTimestamp($t2);

print $dt->format('Y-m-d H:i:s.u') . "\n";
```

See https://3v4l.org/BBBSl
### Expected result

```
2017-04-28 04:02:19.836264
1993-08-30 02:01:09.836264
```
### Actual result

```
2017-04-28 04:02:19.836264
1993-08-30 02:01:09.000000
```

**Comment by mofarrell** [comment]
PHP7 behaves as we do.

**Comment by Orvid** [comment]
PHP7 dropping it is a bug, as-is our dropping it.

**Comment by mofarrell** [comment]
Looks intentional: https://github.com/php/php-src/commit/55626549d81d0feadb1d160be78fcf2b898a48cc

**Comment by DemiMarie** [comment]
What is the workaround?

**Comment by Orvid** [comment]
Well, I was going to say `->modify("+$micro microseconds");` but we don't seem to support it, nor does earlier versions of PHP...

https://3v4l.org/1SBFd

I think we should fix that before we close this as won't fix.

**Comment by jesseschalken** [comment]
I re-ran the 3v4l (https://3v4l.org/BBBSl) and it appears all released versions of 7.1 (7.1.0 to 7.1.4) behaves the same as HHVM.

**Comment by mofarrell** [comment]
Hmm.  I think it is worth closing this issue, and opening one to track adding support for `->modify("+$micro microseconds");`
